### ISSUE TITLE
Allow to send password recovery links using SMS

### ISSUE THREAD

**Issue by kukoarmas** [goauthentik/authentik#9161]
**Password recovery to email may not always be a good idea**

In organizations where corporate email is linked to the IdP, the password recovery email is useless

Of course, we can use an alternate email address for password recovery, but if it's a user's private email, organization has no control over it and user may not take security seriously.
Also, in many organizations, it may not be acceptable by the organization security policy to send password recovery links to external email addresses

**Describe the solution you'd like**

It would be really useful to have something similar to the email stage (sending messages based on templates, etc.) but sending an SMS message instead.
The phone number could be read from the users attributes (just as many of us are doing now with alternate email address for password recovery)

That would allow to:
* Send recovery link to SMS from the admin console (the same way we can send email now)
* Create a recovery flow that users can access directly and, instead of sending email to alternate email address, sends the recovery link to user's phone in SMS message

This functionality could also by used for users' enrollment (first login): Admin creates users, sets phone number, and sends recovery SMS (or just tell the user to recover password to his phone number)

This would be more secure than alternate email address because (normally) phones are more secure than email. And also, in a context of an organization, the organization may have control over user's phone used for password recovery

**Describe alternatives you've considered**

In users that already had enrolled for SMS MFA, we can create a flow that allows for password recovery asking for 2 MFA methods (i.e SMS and TOTP)

For user's enrollment (first login), we are creating user, creating a recovery link, and sending it vie SMS from an external service (using a script that creates the recovery link using authentik API

But it could be much more easy and useful to have the functionality directly in authentik

**Additional context**

**Comment by BeryJu** [comment]
While this is not directly supported, as the recovery is done using flows, you can configure any verification method desired for recovery. You can build a recovery flow that uses an identification stage + an authenticator validation stage to let the user verify themselves via TOTP/SMS TOTP/WebAuthn/etc, which might still fit the use-case

**Comment by kukoarmas** [comment]
Yes, I mentioned that option in the alternatives section. But it's only useful users that already have MFA configured. It's not useful for users' first enrollment.
The use case I'm trying to solve is creating a new user and sending them a password recovery link via SMS, because users will use a email address that is linked to authentik idp

**Comment by Ogoyukin** [comment]
Hi team! 👋

📌 Problem
In previous discussions this feature was removed (or avoided) in favor of dynamic Flows. Change password feature still need a straightforward way to ensure that users must change their password on their next login (e.g., after being created by an admin).

✅ Proposed implementation (compatible with the Flow-based design):
Add a boolean attribute force_password_change to the user object (stored in user.attributes["force_password_change"]).

In the login flow, add a stage or Flow Plan that:

checks if the user has force_password_change: true
if so, redirects them to a change-password Flow
In the change-password Flow, after successful password change:

a simple hook or logic clears the force_password_change flag.
This way:
No need to store a new field in the database
Behavior is still fully driven by Flows
Integrators and admins get an intuitive, practical behavior that's present in most IdPs

What do you think about this way?
Thanks ChatGPT
